quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_format.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::convert::Infallible;
+use std::fmt::Display;
+use std::str::FromStr;
+
+use serde::{Deserialize, Deserializer, Serialize};
+use unwrap_infallible::UnwrapInfallible;
+
+/// Specifies the datetime and unix timestamp formats to use when parsing date strings.
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub enum DateTimeFormat {
+    ISO8601,
+    RFC2822,
+    RCF3339,
+    Strftime {
+        strftime_format: String,
+        with_timezone: bool,
+    },
+    TimestampSecs,
+    TimestampMillis,
+    TimestampMicros,
+}
+
+impl DateTimeFormat {
+    pub fn as_str(&self) -> &str {
+        match self {
+            DateTimeFormat::ISO8601 => "iso8601",
+            DateTimeFormat::RFC2822 => "rfc2822",
+            DateTimeFormat::RCF3339 => "rfc3339",
+            DateTimeFormat::Strftime {
+                strftime_format, ..
+            } => strftime_format,
+            DateTimeFormat::TimestampSecs => "unix_ts_secs",
+            DateTimeFormat::TimestampMillis => "unix_ts_millis",
+            DateTimeFormat::TimestampMicros => "unix_ts_micros",
+        }
+    }
+
+    pub fn is_timestamp(&self) -> bool {
+        matches!(
+            self,
+            DateTimeFormat::TimestampSecs
+                | DateTimeFormat::TimestampMillis
+                | DateTimeFormat::TimestampMicros
+        )
+    }
+}
+
+impl Display for DateTimeFormat {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        f.write_str(self.as_str())
+    }
+}
+
+impl FromStr for DateTimeFormat {
```

Comment:
TryFromStr would be nicer here. Not all strings as valid...

Reply:
Ah, scratch that... actually we accept any string down the line and there are no intermediary date format object. That's a bit lame (from the chrono crate that is).

If there are no blockers, I suggest we ditch chorno and rely on the time crate instead.
`time` makes it possible to first parse a format description.
https://docs.rs/time/0.3.15/time/format_description/fn.parse.html

Also chrono does not seem much maintained and has a bunch of lagging dependencies.